server/session.go

```diff
@@ -12,13 +14,36 @@ type contextKey string
 
 const userContextKey contextKey = "user"
 
+func generateSessionID() (string, error) {
+	randomBytes := make([]byte, 32)
+	_, err := rand.Read(randomBytes)
+	if err != nil {
+		return "", err
+	}
+	return base64.URLEncoding.EncodeToString(randomBytes), nil
+}
+
+type AuthSessionData struct {
```

Comment:
As I said in another comment, you could move these to the `app` package because it's part of our app logic and being persisted in Redis.